### PR TITLE
GuLoader tooling

### PR DESCRIPTION
**Pull request by dstepanic**
Adding a IDAPython script to fix CFG for blog release

### CODE REVIEW

File: tools/guloader/guloader_FixCFG.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    Retrieves all addresses from TinyTracer log file
+    
+    :param file: The path to the TinyTracer log file
+    :return: A list of addresses from TinyTracer log file
+             None if extraction fails or encounters an exception.
+    """
+
+    addrs = list()
+    with open(file, "r") as file:
+        for line in file:
```
Comment (soolidsnake) [inline tools/guloader/guloader_FixCFG.py:17]:
```suggestion
        for line in file:
            try:
                match = re.search(r'([0-9a-fA-F]+)\+([0-9a-fA-F]+);', line)
                if match:
                    result_address = int(match.group(1), 16) + int(match.group(2), 16)
                    addrs.append(result_address)
                else:
                    print(f"Invalid line format: {line.strip()}")
            except ValueError as e:
                print(f"Error processing line: {line.strip()}. {e}")
    return addrs 
```

File: tools/guloader/guloader_FixCFG.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                        result_address = int(hex(address1 + address2),16)
+                        addrs.append(result_address)
+                    except ValueError as e:
+                        print(f"Error processing line: {line.strip()}. {e}")
+                else:
+                    print(f"Invalid format for addresses in line: {line.strip()}")
+            else:
+                print(f"Invalid line format: {line.strip()}")
+
+    return addrs
```
Comment (soolidsnake) [inline tools/guloader/guloader_FixCFG.py:33]:
```suggestion
        for line in file:
            try:
                match = re.search(r'([0-9a-fA-F]+)\+([0-9a-fA-F]+);', line)
                if match:
                    result_address = int(match.group(1), 16) + int(match.group(2), 16)
                    addrs.append(result_address)
                else:
                    print(f"Invalid line format: {line.strip()}")
            except ValueError as e:
                print(f"Error processing line: {line.strip()}. {e}")
    return addrs 
```

Comment (dstepanic) [reply tools/guloader/guloader_FixCFG.py:33]: Thanks for the suggestion! Looks much cleaner @soolidsnake.

**Comment by dstepanic** [comment]
Thanks @soolidsnake for the review. I have now integrated the regex, I tested it out and it's working good. Thanks.

**Review by soolidsnake — APPROVED** [review]
LGTM!

**Pull request merged**